Source/ARTDeviceDetails.h

```diff
@@ -7,7 +7,7 @@ NS_ASSUME_NONNULL_BEGIN
 
 @interface ARTDeviceDetails : NSObject
 
-@property (nonatomic) ARTDeviceId *id;
+@property (nullable, nonatomic) ARTDeviceId *id;
```

Comment:
I'd like to know whether `id` can, in normal use, be `nil`, or whether we're just using `id == nil` to communicate to `-[ARTRest device_nosync]` that the device needs to be regenerated? If the latter, then perhaps we should use a separate flag to represent this state, and keep this non-nullable.

Reply:
I would avoid flags like "wasReset" or "shouldBeRegenerated" and keep built-in mechanism for that - nullability. That states exactly that -  the property "was reset" or "should be reassigned". But in this particular scenario I should probably went with resetting the whole object with `resetDeviceSingleton` and checking that instead. So yeah, `id` shouldn't be `nil` here.

Reply:
I've returned nonnull device `id`. Also I don't check on nil anymore, since all creation of a device singlton are explicit in places where it's needed. Also `reset` method nullifies all properties of the device because example app can't understand `nil` value. Once it's updated `reset` will only clear storage and nullify shared device itself.
@ben-xD @lawrence-forooghian

Reply:
e3cb20eb